crates/block-producer/src/batch_builder/batch.rs

```diff
@@ -88,7 +85,7 @@ impl TransactionBatch {
     #[instrument(target = "miden-block-producer", name = "new_batch", skip_all, err)]
     pub fn new(
         txs: Vec<ProvenTransaction>,
-        found_unauthenticated_notes: BTreeMap<NoteId, MerklePath>,
+        found_unauthenticated_notes: BTreeMap<NoteId, NoteInclusionProof>,
```

Comment:
Related to the previous comments: I don't think `BTreeMap<NoteId, NoteInclusionProof>` would be sufficient to capture all required information. We'll probably need to create a separate struct which contains all additional info (i.e., block headers and MMR proofs).

The questions is what to call this struct. One option is `UnauthenticatedNoteWitness` - but I don't love this name.

Reply:
Ignored for now